src/security/builtin_plugins/authentication/src/authentication.c

```diff
@@ -216,21 +220,19 @@ typedef struct {
 } validity_cb_info;
 
 
+static int compare_relation (const void *va, const void *vb);
+
+const ddsrt_avl_treedef_t relations_treedef =
+  DDSRT_AVL_TREEDEF_INITIALIZER (offsetof (struct IdentityRelation, avlnode), offsetof (struct IdentityRelation, remoteIdentity), compare_relation, 0);
```

Comment:
I had naively expected that identies would be compared on the content of them, but this, combined with the definition of ``compare_relation``, rather says identies are compared by the address at which they are stored. Is that really the intent? It could be ...

(And if it is not the intent, I would suggest using ``DDSRT_AVL_TREEDEF_INITIALIZER_INDKEY``.)

Reply:
No this was not the intend. Will be changed as suggested.